nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ProcessGroupResource.java

```diff
@@ -238,6 +313,47 @@ public Response getProcessGroup(
         return generateOkResponse(entity).build();
     }
 
+
+    /**
+     * Retrieves the Variable Registry for the group with the given ID
+     *
+     * @param groupId the ID of the Process Group
+     * @return the Variable Registry for the group
+     */
+    @GET
+    @Consumes(MediaType.WILDCARD)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{id}/variable-registry")
+    @ApiOperation(value = "Gets a process group's variable registry", response = VariableRegistryEntity.class, authorizations = {
+        @Authorization(value = "Read - /process-groups/{uuid}", type = "")
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = 400, message = "NiFi was unable to complete the request because it was invalid. The request should not be retried without modification."),
+        @ApiResponse(code = 401, message = "Client could not be authenticated."),
+        @ApiResponse(code = 403, message = "Client is not authorized to make this request."),
+        @ApiResponse(code = 404, message = "The specified resource could not be found."),
+        @ApiResponse(code = 409, message = "The request was valid but NiFi was not in the appropriate state to process it. Retrying the same request later may be successful.")
+    })
+    public Response getVariableRegistry(
```

Comment:
Can we add a flag (probably enabled by default) that will include all variables available in this scope (defined in this Process Group and any ancestors)? Also, the response should indicate where each variable is defined.